pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -95,7 +95,24 @@ func (d *DependenciesDistributor) OnAdd(obj interface{}) {
 }
 
 // OnUpdate handles object update event and push the object to queue.
-func (d *DependenciesDistributor) OnUpdate(_, newObj interface{}) {
+func (d *DependenciesDistributor) OnUpdate(oldObj, newObj interface{}) {
+	unstructuredOldObj, err := helper.ToUnstructured(oldObj)
+	if err != nil {
+		klog.Errorf("Failed to transform oldObj, error: %v", err)
+		return
+	}
+
+	unstructuredNewObj, err := helper.ToUnstructured(newObj)
+	if err != nil {
+		klog.Errorf("Failed to transform newObj, error: %v", err)
+		return
+	}
+
+	if !detector.SpecificationChanged(unstructuredOldObj, unstructuredNewObj) {
```

Comment:
Do we need to care about labels?

Reply:
Yes. Labels are commonly used fields.

Reply:
if labels changed, then `!detector.SpecificationChanged(unstructuredOldObj, unstructuredNewObj)` will be `false`.